Firestore/CMakeLists.txt

```diff
@@ -15,6 +15,53 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(firestore C CXX)
 
+option(ASAN "Build with Address sanitizer (mutually exculsive with TSAN)" OFF)
```

Comment:
All your changes here might make more sense in cmake/CompilerOptions.cmake, though they would then apply across all of the repo. (As long as everything defaults to off, maybe that's ok?)

Reply:
I'm okay with that, let's postpone until the next round of review.

Reply:
It's better not to hide the options we support since this file serves as documentation for people trying to figure out how to configure this build to their purposes.

Reply:
@wilhuff Do you think that it would be worthwhile to have these options in the Firebase build as well? If that were the case, I thought about splitting the sanitizer logic from here into `./cmake/Sanitizers.cmake` and including that in both projects. It's a good point that it would make these options non-obvious, so perhaps it would be okay to have the declaration of sanitizer options twice, once for each project. What do you think?